src/lib/zkapps_examples/zkapps_examples.ml

```diff
@@ -518,31 +523,43 @@ let compile :
           ; prevs
           ; uses_lookup
           ; main =
+              (* this is the only change that happens, what's happening here? *)
               (fun main_input ->
+                (* run main *)
                 let { Pickles.Inductive_rule.previous_proof_statements
                     ; public_output = account_update_under_construction
                     ; auxiliary_output
                     } =
                   main main_input
                 in
+                (* do something with the output of main: *)
                 let public_output, account_update_tree =
                   to_account_update account_update_under_construction
                 in
                 { previous_proof_statements
                 ; public_output
+                  (* so that's what's changing, the auxiliary output gets an additional [account_update_tree] *)
                 ; auxiliary_output = (account_update_tree, auxiliary_output)
                 } )
           }
           :: go choices
     in
+
+    (* Recurse with initial state [choices] and [~self].
+       I think at this point [self] is [None].
+    *)
     go (choices ~self)
   in
+
+  (* compile with pickles *)
   let tag, cache_handle, proof, provers =
     Pickles.compile () ?self ?cache ?disk_keys
       ~public_input:(Output Zkapp_statement.typ)
       ~auxiliary_typ:Typ.(Prover_value.typ () * auxiliary_typ)
       ~branches ~max_proofs_verified ~name ~constraint_constants ~choices
   in
+
+  (* wrap pickle prover *)
```

Comment:
In the interest of not adding so much noise to the code in, renaming `go` to `wrap_provers` or `convert_provers_to_return_account_update_call_forests` or some such might make it clear what's happening.